src/test/rustdoc/check-source-code-urls-to-def.rs

```diff
@@ -40,5 +42,9 @@ pub fn foo(a: u32, b: &str, c: String, d: Foo, e: bar::Bar, f: source_code::Sour
 
 // @has - '//a[@href="../../src/foo/auxiliary/source-code-bar.rs.html#14-16"]' 'bar::sub::Trait'
 // @has - '//a[@href="../../src/foo/auxiliary/source-code-bar.rs.html#14-16"]' 'Trait'
-pub fn foo2<T: bar::sub::Trait, V: Trait>(t: &T, v: &V) {
+pub fn foo2<T: bar::sub::Trait, V: Trait>(t: &T, v: &V, b: bool) {
 }
+
+// @has - '//a[@href="../../foo/primitive.bool.html"]' 'bool'
+#[doc(primitive = "bool")]
+mod whatever {}
```

Comment:
This seems unnecessarily complicated - can you check it links to doc.rust-lang.org instead, so we're testing how it behaves when there's no doc(primitive) available? That should be the case for literally every crate other than libstd and core.

Feel free to add a separate test for that if you want to keep this one, but I do think it's important to have.

Reply:
I think I'll add a second test so both cases are covered.